src/common.c

```diff
@@ -264,6 +265,20 @@ const char *ofi_straddr(char *buf, size_t *len,
 	case FI_ADDR_MLX:
 		size = snprintf(buf, *len, "fi_addr_mlx://%p", addr);
 		break;
+	case FI_ADDR_IB_UD:
+		memset(str, 0, sizeof(str));
+		for (i = 0; i < 4; i++)
+			sprintf(&str[i * 8], "%08x",
+				htonl(*(uint32_t *)((uint8_t *)addr + i * 4)));
```

Comment:
Use inet_ntop() to convert a GID into a printable address.  A GID has an IPv6 format.